src/snowflake/snowpark/session.py

```diff
@@ -319,6 +317,19 @@ def close(self) -> None:
             finally:
                 _remove_session(self)
 
+    @property
+    def sql_simplifier_enabled(self) -> bool:
+        return self._sql_simplifier_enabled
+
+    @sql_simplifier_enabled.setter
+    def sql_simplifier_enabled(self, value: bool) -> None:
+        """Set to ``True`` to use the SQL simplifier.
+        The generated SQLs from ``DataFrame`` transformations would have fewer layers of nested queries if the SQL simplifier is enabled."""
+        self._conn._telemetry_client.send_sql_simplifier_telemetry(
```

Comment:
Should we also check `telemetry_enabled` here? If `telemetry_enabled` is False, we shouldn't send the telemetry.

Reply:
This method itself checks the whether telemetry is enabled.